src/Helpers.cpp

```diff
@@ -0,0 +1,61 @@
+#include "Helpers.hpp"
+
+using namespace radar_base;
+
+void helpers::iterateOverSweepsAndApplyFunction(Radar const& radar_echo,
+    base::Angle const& yaw_correction,
+    helpers::sweepFunction const& lambda_function)
+{
+    double angle = (radar_echo.start_heading).getRad() + yaw_correction.getRad();
+    if (angle < 0) {
+        angle = 2 * M_PI + angle;
+    }
+
+    int start_angle_unit = round(angle / abs(radar_echo.step_angle.getRad()));
+    int angles_in_a_frame = round(2 * M_PI / abs(radar_echo.step_angle.getRad()));
+    int signal = abs(radar_echo.step_angle.getRad()) / radar_echo.step_angle.getRad();
+    for (int current_angle = 0;
+         current_angle < static_cast<int>(radar_echo.sweep_timestamps.size());
+         current_angle++) {
+
+        int sweep_global_angular_position =
+            (start_angle_unit + current_angle * signal) % angles_in_a_frame;
+        if (sweep_global_angular_position < 0) {
+            sweep_global_angular_position += angles_in_a_frame;
+        }
+        lambda_function(radar_echo, current_angle, sweep_global_angular_position);
+    }
+}
+
+void helpers::updateLookUpTable(int num_angles,
+    int sweep_size,
+    float beam_width,
+    int output_image_size,
+    std::unique_ptr<radar_base::EchoToImageLUT>& m_lut)
+{
+    m_lut.reset(new radar_base::EchoToImageLUT(num_angles,
+        sweep_size,
+        beam_width,
+        output_image_size));
+}
+
+bool helpers::updateLookUpTableIfNeeded(int num_angles,
```

Comment:
I think a better way would be save the LUT parameters within the LUT itself

```cpp
class EchoToImageLUT {
public:
    int num_angles_
    int sweep_size_;
    float beam_width_;
    int window_size_;
    
    EchoToImageLUT(int num_angles,
        int sweep_size,
        float beam_width,
        int window_size)
        : num_angles_(num_angles)
        , sweep_size_(sweep_size)
        , beam_width_(beam_width)
        , window_size_(window_size)
    {
        setup(sweep_size, window_size, num_angles, beam_width);
    }
};

```
and then have a member method telling if some parameters set match this LUT parameters. 

Another thing is that `range` does not influence LUT result so why are we checking for range changes to decide weather we should reset the LUT or not?